inc/Engine/Optimization/AssetsLocalCache.php

```diff
@@ -109,4 +113,55 @@ protected function write_file( $content, $file ) {
 
 		return rocket_put_content( $file, $content );
 	}
+
+	/**
+	 * Check if this link HTML has integrity attribute or not?
+	 *
+	 * @since 3.7.5
+	 *
+	 * @param string $asset Link HTML to be tested.
+	 *
+	 * @return array|false Matched array with integrityhashmethod, integrityhash keys.
+	 */
+	private function has_integrity( $asset ) {
+		if ( ! preg_match( '#\s*integrity\s*=[\'"](?<integrityhashmethod>.*)-(?<integrityhash>.*)[\'"]#Ui', $asset, $integrity_matches ) ) {
+			return false;
+		}
+
+		if ( ! isset( $integrity_matches['integrityhashmethod'] ) || ! isset( $integrity_matches['integrityhash'] ) ) {
+			return false;
+		}
+
+		return $integrity_matches;
+	}
+
+	/**
+	 * Validate the integrity attribute if the content matches with the hashed integrity value.
+	 *
+	 * @param array $asset_matched the matched array which has 0, url keys.
+	 *
+	 * @return bool|string
+	 */
+	public function validate_integrity( $asset_matched ) {
+		$integrity_matches = $this->has_integrity( $asset_matched[0] );
+
+		if ( false === $integrity_matches ) {
+			return $asset_matched[0];
+		}
+
+		// validate the hash algorithm.
+		if ( ! in_array( $integrity_matches['integrityhashmethod'], hash_algos(), true ) ) {
+			return false;
+		}
+
+		$content      = $this->get_content( $asset_matched['url'], false );
+		$content_hash = base64_encode( hash( $integrity_matches['integrityhashmethod'], $content, true ) );// phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+
+		if ( $integrity_matches['integrityhash'] !== $content_hash ) {
+			return false;
+		}
+
+		return str_replace( $integrity_matches[0] . '', '', $asset_matched[0] );
```

Comment:
Why is there a need for a concatenation with an empty string `$integrity_matches[0] . ''`?